lib/mock_data/default_data_yaml/default_users_and_roles.yml

```diff
@@ -155,8 +155,8 @@ teachers:
 
   teacher_5:
     login: teacher_with_no_class
-    first_name: Teacher_with_no_class
-    last_name: Teacher_with_no_class
+    first_name: TeacherWithNoClass
```

Comment:
No _ is not still valid. Should it be?

The new regex:
 +  @@name_regex      = /\A[\p{L}\d]+\z/          # Unicode, permissive 

is actually matching on a smaller subset of the ASCII, since _ is not considered to be in the "Letter Category", which is what \p{L} matches. In fact I had to add \d which was previously matched as a word character by the \w class. We can add additional ASCII characters (e.g. those allowed by the login_regex) but we cannot really make an equivalent to the old name_regex which was based on exclusion of characters: /\A[^[:cntrl:]\\<>\/&]*\z/  

What are your thoughts on this?